From 5b99652d67eda87cf69b623ce02abba057c851d8 Mon Sep 17 00:00:00 2001
From: Axel Gembe <ago@seca-online.de>
Date: Wed, 11 Jun 2008 09:56:02 +0200
Subject: [PATCH] fix double free in print preview

The printout gets freed in the wxPrintPreviewBase destructor as well as in
wxPreviewFrame::OnCloseWindow. The problem is that almost none of the code in
prntbase.cpp uses wxDELETE, which manifests itself in this problem, because
while OnCloseWindow does free the printout, it does not set the pointer to NULL,
which causes the destructor to double-free the pointer. I really think
everything should be converted to use wxDELETE in the long run.

This patch is against the wxWidgets SVN branch WX_2_8_BRANCH.

Signed-off-by: Axel Gembe <ago@seca-online.de>
---
 src/common/prntbase.cpp |   11 +----------
 1 files changed, 1 insertions(+), 10 deletions(-)

diff --git a/src/common/prntbase.cpp b/src/common/prntbase.cpp
index 43d3a36..d9a1fa9 100644
--- a/src/common/prntbase.cpp
+++ b/src/common/prntbase.cpp
@@ -1237,16 +1237,7 @@ void wxPreviewFrame::OnCloseWindow(wxCloseEvent& WXUNUSED(event))
     if (m_windowDisabler)
         delete m_windowDisabler;
 
-    // Need to delete the printout and the print preview
-    wxPrintout *printout = m_printPreview->GetPrintout();
-    if (printout)
-    {
-        delete printout;
-        m_printPreview->SetPrintout(NULL);
-        m_printPreview->SetCanvas(NULL);
-        m_printPreview->SetFrame(NULL);
-    }
-    delete m_printPreview;
+    wxDELETE(m_printPreview);
 
     Destroy();
 }
-- 
1.5.4.2

